scanpipe/models.py

```diff
@@ -633,6 +638,14 @@ def delete(self, *args, **kwargs):
 
         return super().delete(*args, **kwargs)
 
+    def mark_for_deletion(self):
+        self.update(is_marked_for_deletion=True)
+
+    def delete_in_background(self):
+        # Mark the project for deletion and enqueue background deletion task
+        self.mark_for_deletion()
```

Comment:
I think we can put `self.update(is_marked_for_deletion=True)` directly here, no need for the `mark_for_deletion` method yet since it's a one-liner that is only used in one place.